### PR TITLE
Handle str and pathlib.Path consistently in image_io

### PR DESCRIPTION
**Pull request by K-Meech**
## Description

**What is this PR**

- [ ] Bug fix
- [X] Addition of a new feature
- [ ] Other

**Why is this PR needed?**
The load/save functions in `image_io` currently only specify `str` filepaths in their docstrings, although some can also accept `pathlib.Path`. 

**What does this PR do?**
This PR ensures that all load/save functions in `image_io` consistently handle both `str` and `pathlib.Path`. It favours using `pathlib.Path` internally wherever possible.

## References

Closes https://github.com/brainglobe/brainglobe-utils/issues/47

## How has this PR been tested?

`image_io` tests were updated so that all load/save functions are tested with both `pathlib.Path` and `str`. 

## Is this a breaking change?

No

## Does this PR require an update to the documentation?

All relevant `image_io` docstrings were updated.

## Checklist:

- [X] The code has been tested locally
- [X] Tests have been added to cover all new functionality (unit & integration)
- [X] The documentation has been updated to reflect any changes
- [X] The code has been formatted with [pre-commit](https://pre-commit.com/)

### CODE REVIEW

File: brainglobe_utils/image_io/load.py [hunk 1]
```diff
@@ -112,7 +112,7 @@ def load_any(
             sort=sort_input_file,
             n_free_cpus=n_free_cpus,
         )
-    elif src_path.endswith((".tif", ".tiff")):
+    elif src_path.suffix == ".tif" or src_path.suffix == ".tiff":
```
Comment (willGraham01) [inline brainglobe_utils/image_io/load.py:115]:
```suggestion
    elif src_path.suffix in [".tif", ".tiff"]:
```

I am a mathematician so an naturally going to suggest the set-like notation and slightly fewer keystrokes :sweat_smile:

File: brainglobe_utils/image_io/load.py [hunk 2]
```diff
@@ -121,10 +121,10 @@ def load_any(
             z_scaling_factor,
             anti_aliasing=anti_aliasing,
         )
-    elif src_path.endswith(".nrrd"):
+    elif src_path.suffix == ".nrrd":
         logging.debug("Data type is: nrrd")
         img = load_nrrd(src_path)
-    elif src_path.endswith((".nii", ".nii.gz")):
+    elif src_path.suffix == ".nii" or src_path.suffix == ".nii.gz":
```
Comment (willGraham01) [inline brainglobe_utils/image_io/load.py:127]:
```suggestion
    elif src_path.suffix in [".nii", ".nii.gz"]:
```

File: brainglobe_utils/image_io/save.py [hunk 3]
```diff
@@ -125,16 +130,16 @@ def save_any(img_volume, dest_path):
     dest_path = Path(dest_path)
 
     if dest_path.is_dir():
-        to_tiffs(img_volume, str(dest_path / "image"))
+        to_tiffs(img_volume, dest_path / "image")
 
     elif dest_path.suffix == ".tif" or dest_path.suffix == ".tiff":
```
Comment (willGraham01) [inline brainglobe_utils/image_io/save.py:135]:
```suggestion
    elif dest_path.suffix in [".tif", ".tiff"]:
```

File: brainglobe_utils/image_io/load.py [hunk 4]
```diff
@@ -86,9 +86,9 @@ def load_any(
     np.ndarray
         The loaded brain.
     """
-    src_path = str(src_path)
+    src_path = Path(src_path)
```
Comment (willGraham01) [inline brainglobe_utils/image_io/load.py:89]: It's times like this when I wish Python's type-hinting actually enforced casting when passing inputs :joy:

**Review by willGraham01 — APPROVED** [review]

File: brainglobe_utils/image_io/load.py (hunk 4 above)
Comment (K-Meech) [reply brainglobe_utils/image_io/load.py:89]: Me too! :sweat_smile:

**Pull request merged**